text/0000-minimum-docs-lifecycle.md

```diff
@@ -0,0 +1,62 @@
+# Meta
+[meta]: #meta
+- Name: Define a minimum standard for docs in order to ship the lifecycle
+- Start Date: 4/7/2021
+- Author(s): natalieparellano
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The project has made improving our documentation a priority for 2021. In order to help deliver on this promise, this RFC proposes defining a minimum standard for documentation that should be met before we're able to ship new versions of the lifecycle.
+
+# Definitions
+[definitions]: #definitions
+
+Docs repo: https://github.com/buildpacks/docs
+
+Migration guides: https://github.com/buildpacks/docs/tree/main/content/docs/reference/spec/migration
+
+Lifecycle-specific content: https://github.com/buildpacks/docs/tree/main/content/docs/concepts/components/lifecycle
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this? Better docs!
+- What use cases does it support? Every persona will be happier about this.
+- What is the expected outcome? Better docs!
+
+# What it is
+[what-it-is]: #what-it-is
+
+Today, our minimum standard for docs gating the implementation team's ability to ship the lifecycle is largely non-existent. We do require that the spec has been released (for which there are release notes), and we also provide release notes for the lifecycle. We've lately been trying to have migration guides at least in PR status to the docs repo. However the migration guides often do not contain enough information to sufficiently explain a feature, because "core" documentation is missing - see [exec.d on Windows](https://buildpacks.io/docs/reference/spec/migration/buildpack-api-0.5-0.6/#execd-on-windows) which explains that Windows is now supported but doesn't link to any further documentation to understand exec.d, because said documentation does not exist.
+
+Proposal: For each new version of the lifecycle, we should create a `release/lifecycle/<lifecycle version>` milestone and branch on the docs repo (see [docs release process](https://github.com/buildpacks/docs/blob/main/RELEASE.md#lump-changes) for more information on this process). Prior to shipping the lifecycle, `release/lifecycle/<lifecycle version>` should contain:
+* Migration guides for the new APIs implemented.
+* New content in the lifecycle-specific docs (or wherever is relevant) such that every new feature in the lifecycle is fully explained in the docs (ideally, this content could be linked to from the migration guides).
```

Comment:
This seems like a good idea. But I am not sure what the difference is between the process proposed here and what exists in the docs release process, which already outlines how to create branches and milestones? We just need to be better about adding a comprehensive set of issue to the milestone?

If we are struggling to do this with the existing tooling do we need more automation or processes to actually make it happen. Strawman examples:
1. A script that adds a new supported API to the lifecycle and simultaneously generates migration guide issue that is added to the relevent milestone.
2. A org level projects for each API version, to give a birds eye view all changes related to an API?

Reply:
I think automation could be helpful. I guess the intent here is more to create a forcing function to actually do these things.

Reply:
(With no "one" or "thing" holding us accountable today we are just doing the best we can in the time that we have and shipping the lifecycle when the code is ready. This is like, let's draw a line in the sand and change our behavior :))